crates/polars-ops/src/chunked_array/list/namespace.rs

```diff
@@ -265,75 +265,15 @@ pub trait ListNameSpaceImpl: AsList {
-                DataType::Int8 => {
-                    let ca = s_ref.i8()?;
-                    let filtered_ca = ca.filter_with_func(&lambda_expressions)?;
-
-                    Ok(filtered_ca.into_series())
-                },
-                #[cfg(feature = "dtype-i16")]
-                DataType::Int16 => {
-                    let ca = s_ref.i16()?;
-                    let filtered_ca = ca.filter_with_func(&lambda_expressions)?;
-                    Ok(filtered_ca.into_series())
-                },
-                DataType::Int32 => {
-                    let ca = s_ref.i32()?;
-                    let filtered_ca = ca.filter_with_func(&lambda_expressions)?;
-                    Ok(filtered_ca.into_series())
-                },
-                DataType::Int64 => {
-                    let ca = s_ref.i64()?;
-                    let filtered_ca = ca.filter_with_func(&lambda_expressions)?;
-                    Ok(filtered_ca.into_series())
-                },
-                DataType::Float32 => {
-                    let ca = s_ref.f32()?;
-                    let filtered_ca = ca.filter_with_func(&lambda_expressions)?;
-                    Ok(filtered_ca.into_series())
-                },
-                DataType::Float64 => {
-                    let ca = s_ref.f64()?;
-                    let filtered_ca = ca.filter_with_func(&lambda_expressions)?;
-                    Ok(filtered_ca.into_series())
-                },
-                DataType::String => {
-                    let ca = s_ref.str()?;
-                    let filtered_ca = ca.filter_with_func(&lambda_expressions)?;
-                    Ok(filtered_ca.into_series())
-                },
-                DataType::List(_) => {
-                    let ca = s_ref.list()?;
-                    let filtered_ca = ca.filter_with_func(&lambda_expressions)?;
-                    Ok(filtered_ca.into_series())
-                },
-                DataType::Binary => {
-                    let ca = s_ref.binary()?;
-                    let filtered_ca = ca.filter_with_func(&lambda_expressions)?;
-                    Ok(filtered_ca.into_series())
-                },
-                _ => {
-                    polars_bail!(
-                        ComputeError: "filter_with_func not implemented for type: {:?}",
-                        s_ref.dtype()
-                    );
-                },
-            };
-            filtered_inner
+            s_ref.filter(lambda_expressions.eval(s_ref, true)?.bool()?)
         })?;
 
         Ok(filtered)
     }
 
-    fn lst_transform(
-        &self,
-        lambda_expressions: Arc<LambdaExpression>,
-    ) -> PolarsResult<ListChunked> {
+    fn lst_transform(&self, lambda_expression: Arc<LambdaExpression>) -> PolarsResult<ListChunked> {
         let ca = self.as_list();
-        ca.try_apply_amortized(|s| s.as_ref().transform(&lambda_expressions))
+        ca.try_apply_amortized(|s| lambda_expression.eval(s.as_ref(), true))
```

Comment:
Transform function have two arguments - first is element that it's transforms and second it's index